beacon_node/http_api/src/block_id.rs

```diff
@@ -211,6 +211,38 @@ impl BlockId {
             }
         }
     }
+
+    pub fn is_finalized<T: BeaconChainTypes>(&self, chain: &BeaconChain<T>) -> bool {
+        match &self.0 {
+            CoreBlockId::Head => false,
+            CoreBlockId::Genesis => true,
+            CoreBlockId::Finalized => true,
+            CoreBlockId::Justified => false,
+            CoreBlockId::Slot(slot) => {
+                let finalized_epoch = chain
+                    .canonical_head
+                    .cached_head()
+                    .finalized_checkpoint()
+                    .epoch;
+                let block_epoch = slot.epoch(T::EthSpec::slots_per_epoch());
+                block_epoch <= finalized_epoch
+            }
+            CoreBlockId::Root(root) => {
+                let finalized_epoch = chain
+                    .canonical_head
+                    .cached_head()
+                    .finalized_checkpoint()
+                    .epoch;
+                let block_epoch = chain
+                    .get_blinded_block(root)
+                    .unwrap()
+                    .unwrap()
+                    .slot()
+                    .epoch(T::EthSpec::slots_per_epoch());
+                block_epoch <= finalized_epoch
```

Comment:
When an epoch is finalized it only finalizes the _0th slot_ of the epoch and all prior slots. E.g. if epoch 2 is finalized then slots `<= 64` are finalized, but e.g. slot 65 (which is part of epoch 2) is not.

Reply:
I had no idea 😅
Fixed it (I think) in [this commit](https://github.com/sigp/lighthouse/pull/3753/commits/d4fb8e102643205a66199dab77a0fa89cdb9b723).